SmoothCam/source/arrow_fixes.cpp

```diff
@@ -183,9 +187,13 @@ static void mUpdateArrowFlightPath(RE::Projectile* arrow) {
 }
 
 bool ArrowFixes::Attach() {
+	const auto config = Config::GetCurrentConfig();
 	detArrowFlightPath = eastl::make_unique<UpdateArrowFlightPathDetour>(g_Offsets->UpdateFlightPath, mUpdateArrowFlightPath);
-	if (!detArrowFlightPath->Attach())
-		FatalError(L"Failed to place detour on target function(ArrowFixes::UpdateFlightPath), this error is fatal.");
+	
+	if (config->useProjectileFixes) {
```

Comment:
Since you do this check inside the `mUpdateArrowFlightPath` function, you don't need to disable hooking here (though you still can, if you need to). If you keep this, users that start the game with arrow fixes disabled will just have to restart the game for the fix to re-enable when the setting is changed. If you do want to keep it like this, just add a little note in the MCM description for this setting telling users they might need to restart their game when enabling it.

Otherwise this is all working! Let me know which option above you'd like to do and we can get this merged.

Reply:
I updated the commit with the removing the check in ArrowFixes::Attach(), tested locally and works as expected.